build-logic/convention/src/main/kotlin/AndroidFeatureConventionPlugin.kt

```diff
@@ -28,6 +28,10 @@ class AndroidFeatureConventionPlugin : Plugin<Project> {
             pluginManager.apply {
                 apply("nowinandroid.android.library")
                 apply("nowinandroid.android.hilt")
+                // Serialization is used for type-safe navigation.
+                // TODO: Use the plugin ID from the version catalog when
+                //  https://github.com/gradle/gradle/issues/15383# is resolved
+                apply("org.jetbrains.kotlin.plugin.serialization")
```

Comment:
I'm not sure this comment is needed here, otherwise we should also comment all these

`add("implementation", libs.findLibrary("...").get())`

since they are not technically "type-safe".
Also, we already do exactly that in other build-logic plugins:

https://github.com/android/nowinandroid/blob/afad1b9ac7e3d5efc798e786bd6548229a55735c/build-logic/convention/src/main/kotlin/AndroidApplicationComposeConventionPlugin.kt#L26

https://github.com/android/nowinandroid/blob/afad1b9ac7e3d5efc798e786bd6548229a55735c/build-logic/convention/src/main/kotlin/AndroidApplicationConventionPlugin.kt#L33-L36